src/data/clients/ror-v2-client.ts

```diff
@@ -0,0 +1,240 @@
+/**
+ * Types for ROR API V2 responses and entities
+ */
+export interface RORV2Organization {
+  id: string;
+  names: Name[];
+  status: "active" | "withdrawn" | "deprecated";
+  established: number;
+  locations: Location[];
+  types: string[];
+  email_address: string | null;
+  ip_addresses: string[];
+  aliases: string[];
+  acronyms: string[];
+  links: Link[];
+  matches: string[] | null;
+  labels: Label[];
+  relationships: Relationship[];
+  references: Reference[];
+  external_ids: ExternalId[];
+}
+
+interface Link {
+  type: string;
+  value: string;
+}
+
+interface ExternalId {
+  type: string;
+  all: string[];
+}
+
+interface Name {
+  lang: string;
+  types: string[];
+  value: string;
+}
+
+interface Location {
+  geonames_details: {
+    country_code: string;
+    country_name: string;
+    // continent_code: string;
+    // continent_name: string;
+    // country_subdivision_code: string;
+    // country_subdivision_name: string;
+  };
+  geonames_id: string;
+}
+
+
+interface Label {
+  label: string;
+  iso639: string;
+}
+
+interface Relationship {
+  type: string;
+  id: string;
+  label: string;
+}
+
+
+interface Reference {
+  id: string | null;
+  agency: string;
+  source: string | null;
+}
+
+export interface RORV2SearchResponse {
+  items: RORV2Organization[];
+  number_of_results: number;
+  meta: {
+    types: RORFacet[];
+    countries: RORFacet[];
+  };
+}
+
+export interface RORFacet {
+  id: string;
+  title: string;
+  count: number;
+}
+
+
+export interface RORV2Error extends Error {
+  status?: number;
+  statusText?: string;
+}
+
+/**
+ * Configuration options for the ROR API client
+ */
+export interface RORV2ClientConfig {
+  baseUrl?: string;
+  timeout?: number;
+}
+
+/**
+ * Search parameters for querying organizations
+ */
+export interface RORV2SearchParams {
+  query?: string;
+  page?: number;
+  types?: string | string[];
+  countries?: string | string[];
+}
+
+/**
+ * ROR API V2 Client class
+ */
+export class RORV2Client {
+  private baseUrl: string;
+  private timeout: number;
+
+  constructor(config: RORV2ClientConfig = {}) {
+    this.baseUrl = config.baseUrl || 'https://api.ror.org/v2';
```

Comment:
Should the default API URL be an environment variable?

Reply:
It could.....But in reality, we can probably stick with the default since this is keyed to the ror v2 api.